OpenSim/Common/ComponentList.h

```diff
@@ -168,6 +168,10 @@ class ComponentListIterator :
     bool operator==(const ComponentListIterator& iter) const {
         return _node == &*iter;
     }
+    /// Check for equality using method rather than operator to support scripting
+    bool equals(const ComponentListIterator& iter) const {
```

Comment:
> My understanding of the syntax is that the method is just invoking the operator

Oh, I see. Yes, that's probably right but very confusing and not at all obvious because there are so many potential overloads for `==`. To make it clear, the `equals()` method should contain the real implementation, the `operator==` should just invoke it.